mlflow/java/src/main/java/org/mlflow/sagemaker/ScoringServer.java

```diff
@@ -0,0 +1,270 @@
+package org.mlflow.sagemaker;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import org.mlflow.mleap.MLeapLoader;
+import org.mlflow.models.Model;
+import spark.Request;
+import spark.Response;
+import spark.Service;
+
+/** A RESTful webserver for {@link Predictor Predictors} that runs on the local host */
+public class ScoringServer {
+  public static final String RESPONSE_KEY_ERROR_MESSAGE = "Error";
+
+  public static final int HTTP_RESPONSE_CODE_SERVER_ERROR = 500;
+  public static final int HTTP_RESPONSE_CODE_SUCCESS = 200;
+  public static final int DEFAULT_PORT = 5001;
+
+  private enum RequestContentType {
+    Csv("text/csv"),
+    Json("application/json"),
+    Invalid("invalid");
+
+    private final String value;
+    private static final Map<String, RequestContentType> BY_VALUE_MAP = new HashMap<>();
+
+    static {
+      for (RequestContentType inputType : RequestContentType.values()) {
+        BY_VALUE_MAP.put(inputType.value, inputType);
+      }
+    }
+
+    RequestContentType(String value) {
+      this.value = value;
+    }
+
+    public static RequestContentType fromValue(String value) {
```

Comment:
I think enums have a fromValue like thing -- are you just worried about that throwing an exception?

Reply:
Enums support construction from a string, but they don't provide support for mapping undefined strings to a specific enum like `Invalid`.

Reply:
As mentioned, let's see if we can just catch it